provider/pkg/gen/schema.go

```diff
@@ -160,11 +172,21 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 				for _, p := range kind.Properties() {
 					// JSONSchema type includes `$ref` and `$schema` properties, and $ is an invalid character in
 					// the generated names.
+					// TODO: this replacement is going to be deleted once pulumi/pulumi codegen supports masking $-fields.
 					if strings.HasPrefix(p.name, "$") {
 						p.name = "t_" + p.name[1:]
 					}
 					objectSpec.Properties[p.name] = genPropertySpec(p, kind.canonicalGV, kind.kind)
-					//objectSpec.Required = append(objectSpec.Required, p.name)
+				}
+
+				if kind.IsNested() {
```

Comment:
I'm not _quite_ sure that's the right way to fill required properties, but this combination gives zero changes compared to the old SDK. Notice that it gives **a lot** of changes in the Go SDK. Do you feel those are valid?

Reply:
Yeah, that looks right to me. Of the properties I checked, it looks like they were required properties that were mistakenly treated as optional before.